ADAL/src/ui/ADWebAuthController.m

```diff
@@ -290,6 +290,17 @@ - (BOOL)webAuthShouldStartLoadRequest:(NSURLRequest *)request
         return NO;
     }
     
+    // redirecting to non-https url is not allowed
+    if ([requestURL hasPrefix: @"http"] && ![requestURL hasPrefix: @"https"])
+    {
```

Comment:
Should we just put 
if (![requestURL hasPrefix: @"https"])
and move the checking to the end of this function?

There are some subtle difference :P